server/internal/infrastructure/aws/file.go

```diff
@@ -0,0 +1,276 @@
+	}
+
+	var err error
+	u, _ = url.Parse(baseURL)
+	if err != nil {
+		return nil, rerror.NewE(i18n.T("invalid base URL"))
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(),
+		config.WithRegion(region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKeyID, secretAccessKey, "")),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &fileRepo{
+		bucketName:   bucketName,
+		baseURL:      u,
+		cacheControl: cacheControl,
+		s3Client:     s3.NewFromConfig(cfg),
+	}, nil
+}
+
+func (f *fileRepo) ReadAsset(ctx context.Context, u string, fn string) (io.ReadCloser, error) {
+	p := getS3ObjectPath(u, fn)
+	if p == "" {
+		return nil, rerror.ErrNotFound
+	}
+	return f.read(ctx, p)
+}
+
+func (f *fileRepo) GetAssetFiles(ctx context.Context, u string) ([]gateway.FileEntry, error) {
+	p := getS3ObjectPath(u, "")
+	var fileEntries []gateway.FileEntry
+	paginator := s3.NewListObjectsV2Paginator(f.s3Client, &s3.ListObjectsV2Input{
+		Bucket: aws.String(f.bucketName),
+		Prefix: aws.String(p),
+	})
+
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			return nil, rerror.ErrInternalBy(err)
+		}
+		for _, obj := range output.Contents {
+			fe := gateway.FileEntry{
+				Name: strings.TrimPrefix(lo.FromPtr(obj.Key), p),
+				Size: obj.Size,
+			}
+			fileEntries = append(fileEntries, fe)
+		}
+	}
+
+	if len(fileEntries) == 0 {
+		return nil, gateway.ErrFileNotFound
+	}
+
+	return fileEntries, nil
+}
+
+func (f *fileRepo) UploadAsset(ctx context.Context, file *file.File) (string, int64, error) {
+	if file == nil {
+		return "", 0, gateway.ErrInvalidFile
+	}
+	if file.Size >= fileSizeLimit {
+		return "", 0, gateway.ErrFileTooLarge
+	}
+
+	fileUUID := newUUID()
+	p := getS3ObjectPath(fileUUID, file.Name)
+	if p == "" {
+		return "", 0, gateway.ErrInvalidFile
+	}
+	size, err := f.upload(ctx, p, file.Content)
+	if err != nil {
+		return "", 0, rerror.ErrInternalBy(err)
+	}
+
+	return fileUUID, size, nil
+}
+
+func (f *fileRepo) DeleteAsset(ctx context.Context, u string, fn string) error {
+	p := getS3ObjectPath(u, fn)
+	if p == "" {
+		return gateway.ErrInvalidFile
+	}
+	return f.delete(ctx, p)
+}
+
+func (f *fileRepo) GetURL(a *asset.Asset) string {
+	return getURL(f.baseURL.String(), a.UUID(), a.FileName())
+}
+
+func (f *fileRepo) IssueUploadAssetLink(ctx context.Context, filename, contentType string, expiresAt time.Time) (string, string, error) {
+	uuid := newUUID()
+	p := getS3ObjectPath(uuid, filename)
+	if p == "" {
+		return "", "", gateway.ErrInvalidFile
+	}
+
+	presignClient := s3.NewPresignClient(f.s3Client)
+	uploadURL, err := presignClient.PresignPutObject(ctx, &s3.PutObjectInput{
+		Bucket:       aws.String(f.bucketName),
+		CacheControl: aws.String(f.cacheControl),
+		Key:          aws.String(p),
+		ContentType:  aws.String(contentType),
+	}, func(options *s3.PresignOptions) {
+		options.Expires = time.Minute * 15
```

Comment:
```suggestion
		options.Expires = expires
```

Define it as constant: `const expries = time.Minute * 15`

Reply:
Done!